### PR TITLE
Draggable columns

### PR DESCRIPTION
**Pull request by LucasLeandro1204**
Hey, now we can drag the columns.

I never worked with VueX so I'm probably doing shit here HAUahuhua

The columns order aren't persisting yet, but I'm thinking if it's better to add an order prop to each column, instead of the way I'm doing right now.

### CODE REVIEW

**Comment by LucasLeandro1204** [comment]
I added an editorconfig file too

**Comment by Meadowcottage** [comment]
This is awesome! 😄 
Few changes needed though. 
 - Change column header icon to be drag region rather than whole toolbar
 - Port this to a individual module (Later need to add this functionality to the sidebar)
 - Changes to column order need to be saved in state/config file
 - Also, not looked into `vuedraggable` too much so not sure if possible, add a custom animation/transition for when moving the column. Rather than it just changing the opacity.

File: src/renderer/components/appContent.vue [hunk 1]
```diff
@@ -4,9 +4,7 @@
     <v-content>
       <v-container fluid>
         <v-layout row>
-          <section v-for='i in columns' :key='i'>
-            <appColumn :icon='i.icon' :title='i.title' :columnId='i.id'></appColumn>
-          </section>
+          <appColumnsDraggable></appColumnsDraggable>
```
Comment (LucasLeandro1204) [inline src/renderer/components/appContent.vue]: I didn't like this name

File: src/renderer/store/modules/columns.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import config from '../../../main/lib/config'
 const columnPrefix = 'column-'
 
 const state = {
-  columns: config.get('columns')
+  columns: config.get('columns').sort((a, b) => a.order - b.order)
```
Comment (LucasLeandro1204) [inline src/renderer/store/modules/columns.js]: I don't know if this `sort` is needed,

File: src/renderer/store/modules/columns.js [hunk 3]
```diff
@@ -15,6 +15,13 @@ const mutations = {
   },
   setTitle (state, num, newTitle) {
     state.columns[num].title = newTitle
+  },
+  updateColumnsOrder (state, columns) {
+    state.columns = columns.map((column, i) => {
```
Comment (LucasLeandro1204) [inline src/renderer/store/modules/columns.js]: updating the order

File: src/resources/columns.example.json [hunk 4]
```diff
@@ -3,27 +3,32 @@
     {
       "id": "column-0",
       "icon": "home",
-      "title": "Home"
+      "title": "Home",
+      "order": 0
```
Comment (LucasLeandro1204) [inline src/resources/columns.example.json]: defaults

**Comment by LucasLeandro1204** [comment]
@Meadowcottage Check that out, now there's an order property

**Comment by Meadowcottage** [comment]
Much better! 😄 
Only issue I have found so far in my testing is that the order of the columns is not being saved and refreshing/restarting Muun resets to the default column order.
If you can get that working i'll go ahead and merge this 🙏

**Comment by LucasLeandro1204** [comment]
@Meadowcottage do you have any clue how? I never worked with electron-config, I just use Config.set in store when state changes?

**Comment by Meadowcottage** [comment]
Sadly not. Gonna try testing some stuff and see if I can try and fix that.
Any idea if there's a way to stop the drag item being able to be dragged off the window. EG: Dragged onto the desktop to generate a file 😂

**Comment by LucasLeandro1204** [comment]
I really don't know, even in the browser we can drag it outside =/

**Pull request merged**